relay/pkg/block/push.go

```diff
@@ -70,15 +70,42 @@ func (f *Forwarder) pushHeadersToHostChain(
 
 	if startMod == 0 {
 		// we have a difficulty change first
-		// TODO: implementation
+		logger.Infof(
+			"adding all headers with retarget as there is a difficulty " +
+				"change at the beginning of headers batch",
+		)
+
+		if err := f.addHeadersWithRetarget(headers); err != nil {
+			return fmt.Errorf("could not add headers with retarget: [%v]", err)
+		}
 	} else if startMod > endMod {
 		// we span a difficulty change
-		// TODO: implementation
+		logger.Infof(
+			"adding some headers with retarget as there is a difficulty " +
+				"change in the middle of headers batch",
+		)
+
+		preChangeHeaders, postChangeHeaders := splitBatch(headers, startMod)
+
+		if len(preChangeHeaders) > 0 {
+			if err := f.addHeaders(preChangeHeaders); err != nil {
+				return fmt.Errorf("could not add headers: [%v]", err)
+			}
+		}
+
+		if len(postChangeHeaders) > 0 {
+			if err := f.addHeadersWithRetarget(postChangeHeaders); err != nil {
+				return fmt.Errorf(
+					"could not add headers with retarget: [%v]",
+					err,
+				)
+			}
+		}
 	} else {
 		// no difficulty change
 		logger.Infof(
-			"performing simple headers adding as difficulty doesn't " +
-				"change within headers batch",
+			"simply adding all headers as there is no difficulty change " +
```

Comment:
Instead "simply" (it is unclear what does it actually mean), we can say "adding all headers without retargeting as there..."

Reply:
Sure: https://github.com/keep-network/tbtc/commit/3f435f44610bcb4d040ac5ea88f57f2d070e3ad6